src/Compilers/CSharp/Portable/Binder/DecisionDagBuilder.CheckOrReachability.cs

```diff
@@ -0,0 +1,1433 @@
+                }
+
+                if (!node.Properties.IsDefault)
+                {
+                    var saveMakeEvaluationSequenceOperand = _makeEvaluationSequenceOperand;
+                    BoundPropertySubpattern? property = null;
+
+                    // Given `newPattern`, produce `{ Prop: newPattern }`
+                    _makeEvaluationSequenceOperand = (BoundPattern newPattern) =>
+                    {
+                        // Note: lambda intentionally captures
+                        newPattern = WithInputTypeCheckIfNeeded(newPattern, property!.Pattern.InputType);
+                        ImmutableArray<BoundPropertySubpattern> newSubPatterns = [property.WithPattern(newPattern)];
+
+                        BoundPattern newRecursive = new BoundRecursivePattern(
+                            newPattern.Syntax, declaredType: node.DeclaredType, deconstructMethod: null, deconstruction: default,
+                            properties: newSubPatterns,
+                            isExplicitNotNullTest: false, variable: null, variableAccess: null,
+                            node.InputType, node.NarrowedType, node.HasErrors);
+
+                        if (newPattern.WasCompilerGenerated)
+                        {
+                            newRecursive = newRecursive.MakeCompilerGenerated();
+                        }
+
+                        return saveMakeEvaluationSequenceOperand?.Invoke(newRecursive) ?? newRecursive;
+                    };
+
+                    foreach (BoundPropertySubpattern subPattern in node.Properties)
+                    {
+                        property = subPattern;
+                        VisitPatternAndCombine(node.Syntax, property.Pattern, startOfLeft);
+                    }
+
+                    _makeEvaluationSequenceOperand = saveMakeEvaluationSequenceOperand;
+                }
+
+                _expectingOperandOfDisjunction = saveExpectingOperandOfDisjunction;
+
+                if (_evalSequence.Count - 1 < startOfLeft)
+                {
+                    // Everything was skipped
+                    TryPushOperand(MakeDefaultPattern(node.Syntax, node.InputType));
+                }
+
+                return null;
+            }
+
+            private void VisitPatternAndCombine(SyntaxNode syntax, BoundPattern pattern, int startOfLeft)
+            {
+                int endOfLeft = _evalSequence.Count - 1;
+                int startOfRight = _evalSequence.Count;
+                Visit(pattern);
+                int endOfRight = _evalSequence.Count - 1;
+
+                if (endOfLeft < startOfLeft || endOfRight < startOfRight)
+                {
+                    // Left or right or both are skipped
+                    return;
+                }
+
+                PushBinaryOperation(syntax, endOfLeft, disjunction: _negated);
+            }
+
+            private void TryPushOperandAndCombine(SyntaxNode syntax, BoundPattern pattern, int startOfEval)
```

Comment:
>startOfEval [](http://example.com/codeflow?start=95&length=11)

I do not think the rename is appropriate. The original name was accurately reflecting the meaning of the parameter.